bin/offline-vm-setup.sh

```diff
@@ -29,68 +126,66 @@ autoinstall:
     version: 2
     ethernets:
       enp1s0:
-        dhcp4: yes
-  identity:
-    hostname: $name
-    password: $offline_password
-    username: $offline_username
+        dhcp4: no
+        addresses: [${VM_IP[i]}/24]
+        routes:
+          - to: default
+            via: 192.168.122.1
+  storage:
+    layout:
+      sizing-policy: all
+      name: lvm
+      match:
+        path: /dev/vda
+        size: largest
   ssh:
-    install-server: yes
+    allow-pw: false
+    install-server: true
+  apt:
+    fallback: offline-install
+  user-data:
+    hostname: ${VM_NAME[i]}
+    users:
+    - default
+    - name: demo
+      groups: sudo
+      sudo: ALL=(ALL) NOPASSWD:ALL
+      shell: /bin/bash
+      ssh_authorized_keys: 
+        - $SSHKEY_HUMAN
+        - $SSHKEY_DEMO
 EOF
 }
 
-create_assethost () {
-    name="$1"
-
-    prepare_config "$name"
-
-    # if you want to run the installation manually remove the `--noautoconsole` flag and the ds= part from `--extra-args`
-    sudo virt-install \
-       --name "$name" \
-        --ram 8192 \
-       --disk path=/var/kvm/images/"$name".img,size=100  \
-       --vcpus 4 \
-       --network bridge=br0 \
-       --graphics none \
-       --osinfo detect=on,require=off \
-       --noautoconsole \
-       --location /home/demo/wire-server-deploy/ubuntu.iso,kernel=casper/vmlinuz,initrd=casper/initrd \
-       --extra-args "console=ttyS0,115200n8 autoinstall ds=nocloud-net;s=http://172.16.0.1:3003/$name"
-}
+create_vm () {
+  prepare_config "${VM_NAME[i]}"
 
-create_node () {
-    name="$1"
-
-    prepare_config "$name"
-
-    # if you want to run the installation manually remove the `--noautoconsole` flag and the ds= part from `--extra-args`
-    sudo virt-install \
-        --name "$name" \
-        --ram 8192 \
-        --disk path=/var/kvm/images/"$name".img,size=100 \
-        --vcpus 6 \
-        --network bridge=br0 \
-        --graphics none \
-        --osinfo detect=on,require=off \
-        --noautoconsole \
-        --location /home/demo/wire-server-deploy/ubuntu.iso,kernel=casper/vmlinuz,initrd=casper/initrd \
-        --extra-args "console=ttyS0,115200n8 autoinstall ds=nocloud-net;s=http://172.16.0.1:3003/$name"
+  sudo virt-install \
+    --name "${VM_NAME[i]}" \
+    --ram 8192 \
```

Comment:
not all machines need the same ram, diskspace.

Reply:
https://docs.google.com/document/d/1nX4K5Sdpb66NwsSDE_ppDMtYPvFop3IAzfKhUP2geLY/edit?usp=sharing